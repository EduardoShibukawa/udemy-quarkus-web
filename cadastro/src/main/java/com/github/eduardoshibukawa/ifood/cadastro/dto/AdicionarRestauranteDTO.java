package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class AdicionarRestauranteDTO {
    public String proprietario;

    public String cnpj;

    public String nomeFantasia;

    public LocalizacaoDTO localizacao;
}
