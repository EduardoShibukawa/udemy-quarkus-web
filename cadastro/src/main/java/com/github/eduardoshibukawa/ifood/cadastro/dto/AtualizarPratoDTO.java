package com.github.eduardoshibukawa.ifood.cadastro.dto;

import java.math.BigDecimal;

public class AtualizarPratoDTO {
    public BigDecimal preco;    
}
