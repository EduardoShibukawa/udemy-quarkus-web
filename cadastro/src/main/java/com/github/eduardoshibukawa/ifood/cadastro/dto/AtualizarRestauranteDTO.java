package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class AtualizarRestauranteDTO {
    public String nomeFantasia;
}
