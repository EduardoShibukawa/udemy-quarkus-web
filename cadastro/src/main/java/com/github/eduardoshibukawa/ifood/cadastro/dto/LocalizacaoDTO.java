package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class LocalizacaoDTO {
    
    public Double latitude;

    public Double longitude;
                    
}
