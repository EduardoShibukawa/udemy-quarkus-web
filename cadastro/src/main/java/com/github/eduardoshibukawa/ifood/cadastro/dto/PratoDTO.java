package com.github.eduardoshibukawa.ifood.cadastro.dto;

import java.math.BigDecimal;

public class PratoDTO {

    public Long id;

    public String nome;

    public String descricao;

    public BigDecimal preco;
}
