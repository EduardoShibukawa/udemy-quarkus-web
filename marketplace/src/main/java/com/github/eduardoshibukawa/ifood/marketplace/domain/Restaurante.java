package com.github.eduardoshibukawa.ifood.marketplace.domain;

public class Restaurante {
    
    public Long id;

    public String nome;

    public Localizacao localizacao;   
}
