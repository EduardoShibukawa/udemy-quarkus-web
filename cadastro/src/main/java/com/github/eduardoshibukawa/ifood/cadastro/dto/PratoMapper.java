package com.github.eduardoshibukawa.ifood.cadastro.dto;

import com.github.eduardoshibukawa.ifood.cadastro.domain.Prato;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "cdi")
public interface PratoMapper {

    public Prato toPrato(AdicionarPratoDTO dto);

    public void toPrato(AtualizarPratoDTO dto, @MappingTarget Prato p);

    public PratoDTO toDTO(Prato p);
    
}
