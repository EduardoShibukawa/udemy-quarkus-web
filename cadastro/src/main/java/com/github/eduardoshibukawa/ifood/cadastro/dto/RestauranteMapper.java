package com.github.eduardoshibukawa.ifood.cadastro.dto;

import com.github.eduardoshibukawa.ifood.cadastro.domain.Restaurante;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "cdi")
public interface RestauranteMapper {

    @Mapping(target = "nome", source = "nomeFantasia")
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "dataCriacao", ignore = true)
    @Mapping(target = "dataAtualizacao", ignore = true)
    @Mapping(target = "localizacao.id", ignore = true)
    public Restaurante toRestaurante(AdicionarRestauranteDTO dto);

    @Mapping(target = "nome", source = "nomeFantasia")
    public void toRestaurante(AtualizarRestauranteDTO dto, @MappingTarget Restaurante r);

    @Mapping(target = "nomeFantasia", source = "nome")
    @Mapping(target = "dataCriacao", dateFormat = "dd/MM/yyyy HH:mm:ss")
    public RestauranteDTO toDTO(Restaurante r);
        
}
