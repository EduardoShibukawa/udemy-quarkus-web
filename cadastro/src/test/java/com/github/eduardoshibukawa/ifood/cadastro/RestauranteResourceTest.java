package com.github.eduardoshibukawa.ifood.cadastro;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

import static io.restassured.RestAssured.given;

import java.util.Optional;

import javax.ws.rs.core.Response.Status;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.eduardoshibukawa.ifood.cadastro.configuration.CadastroTestLifecycle;
import com.github.eduardoshibukawa.ifood.cadastro.domain.Restaurante;
import com.github.eduardoshibukawa.ifood.cadastro.dto.AdicionarRestauranteDTO;
import com.github.eduardoshibukawa.ifood.cadastro.dto.AtualizarRestauranteDTO;
import com.github.eduardoshibukawa.ifood.cadastro.util.TokenUtils;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycle.class)
public class RestauranteResourceTest {

    private static final String RESTAURANTES_ENDPOINT = "/restaurantes";
    private static final String RESTAURANTES_CENARIO_1_YML = "restaurantes-cenario-1.yml";
    private String token;

    @BeforeEach
    public void gerarToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null);
    }

    @Test
    @DataSet(RESTAURANTES_CENARIO_1_YML)
    public void testGet() {
        final String resultado = givenWithToken()
            .when().get(RESTAURANTES_ENDPOINT)
            .then()
            .statusCode(Status.OK.getStatusCode())
                .extract().asString();

        Approvals.verifyAsJson(resultado);
    }

    @Test
    @DataSet(RESTAURANTES_CENARIO_1_YML)
    public void testPost() {
        final AdicionarRestauranteDTO dto = new AdicionarRestauranteDTO();

        dto.nomeFantasia = "Saci Lanches";
        dto.proprietario = "Saci";        

        givenJsonContent()
            .body(dto)
            .when().post(RESTAURANTES_ENDPOINT)
            .then()
            .statusCode(Status.CREATED.getStatusCode());

        final Long quantidadeEsperada = 3L;
        final Long quantidadeAtual = Restaurante.count();

        Assert.assertEquals(quantidadeEsperada, quantidadeAtual);
    }

    @Test
    @DataSet(RESTAURANTES_CENARIO_1_YML)
    public void testPut() {
        final Long id = 123L;
        final AtualizarRestauranteDTO dto = new AtualizarRestauranteDTO();

        dto.nomeFantasia = "Capivara Lanches";        

        givenJsonContent()
            .with().pathParam("id", id)
            .body(dto)
            .when().put(RESTAURANTES_ENDPOINT + "/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode());

        final Restaurante atual = Restaurante.findById(id);

        Assert.assertEquals(dto.nomeFantasia, atual.nome);
    }

    @Test
    @DataSet(RESTAURANTES_CENARIO_1_YML)
    public void testDelete() {
        final Long id = 123L;

        givenJsonContent()
            .with().pathParam("id", id)
            .when().delete(RESTAURANTES_ENDPOINT + "/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode());

        final Optional<Restaurante> op = Restaurante.findByIdOptional(id);

        Assert.assertTrue("NÃ£o deve existir restaurante", op.isEmpty());        
    }

    private RequestSpecification givenWithToken() {
        return given()
            .header(new Header("Authorization", "Bearer " + token));
    }

    private RequestSpecification givenJsonContent() {
        return givenWithToken().contentType(ContentType.JSON);
    }    
}
