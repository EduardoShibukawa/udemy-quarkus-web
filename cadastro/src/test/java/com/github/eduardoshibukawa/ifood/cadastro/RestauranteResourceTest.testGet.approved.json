"[{\"id\":123,\"nomeFantasia\":\"Novo Restaurante\",\"proprietario\":\"Id do keyclok\"},{\"id\":1234,\"nomeFantasia\":\"Novo Restaurante 2\",\"proprietario\":\"Id do keyclok 2\"}]"